### PR TITLE
Repo fixups for NET 5 

### LINKED ISSUE #1351: After branching for 5.0, increase the version of WindowsBase to fix incorrect conflict resolution
In Core-Setup, we're trying to pack up 3.0 WPF assets to deliver as a 5.0 WindowsDesktop shared framework, and hit problems with WindowsBase.dll versioning: https://gist.github.com/dagood/5f328e6148acd7cc8ee1afcc6c60f814.

Basically, CoreFX produces a 5.0 WindowsBase.dll, which wins against WPF's 4.8 WindowsBase.dll, causing conflict resolution to favor CoreFX's--which isn't supposed to happen.

This is a tracking issue to make sure it's fixed.

/cc @ericstj @vatsan-madhavan

### LINKED ISSUE #1805: Bring up a branch for .NET 5
Bring up a branch for .NET 5

- [x] Create a branch

- [x] Set up dependency flows

- [ ] Consume an updated netcoreapp5.0 SDK

- [ ] Update TFM in various projects to netcoreapp5.0

- [ ] Update project templates to provide netcoreapp5.0 templates

- [ ] Ensure that WindowsBase conflict resolution works correctly  (see #1351)

### PR DESCRIPTION
**Pull request by vatsan-madhavan**
Fixes #1351, and also includes general fixups that we'd use as a model for getting .NET 3.1 branch up and running. 

There is a twin of this PR corresponding to dotnet-wpf-int at https://dev.azure.com/dnceng/internal/_git/dotnet-wpf-int/pullrequest/2985?_a=overview

Fixes #1805

### CODE REVIEW

**Comment by vatsan-madhavan** [comment]
@ericstj, @dagood, I've validated that WindowsBase conflict resolution works fine now with `AssemblyVersion=5.0.0.0`.

File: eng/WpfArcadeSdk/tools/RuntimeFrameworkReference.targets [hunk 1]
```diff
@@ -25,7 +29,7 @@
   </ItemGroup>
 
   <PropertyGroup>
-    <!-- If TargetFramework is not netcoreapp3.x, then reset RuntimeFrameworkVersion -->
-    <RuntimeFrameworkVersion Condition="!$(TargetFramework.StartsWith('netcoreapp3.'))" />
+    <!-- If TargetFramework is not one of the supported .NET Core versions, then reset RuntimeFrameworkVersion -->
```
Comment (rladuca) [inline eng/WpfArcadeSdk/tools/RuntimeFrameworkReference.targets]: Is the core moniker applicable for 5.0?  I don't care so much about the comments, but are there error/warning messages specific to WPF we need to update?  (Though those are somewhere else, right?)

Comment (vatsan-madhavan) [reply eng/WpfArcadeSdk/tools/RuntimeFrameworkReference.targets]: Error messages like what? The ones in WindowsDesktop SDK or NET SDK rely on `<` or `<=` tests - so no need to update them.  Everywhere else that needed updates to `netcoreapp3.0` related props/targets/*proj have been updated. 

.NET 5.0 is `netcoreapp5.0` - so yeah, I would expect it to have `TargetFrameworkIdentifier==.NETCoreApp`. 

See [this](https://github.com/dotnet/sdk/blob/fe6b50c78426124e34e56ecb7df8b6b86e6e0b48/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.TargetFrameworkInference.targets#L52-L57) in https://github.com/dotnet/sdk: 
```xml
  <!-- Map short name to long name. See earlier comment for example of how to work with identifiers that are not recognized here. -->
  <PropertyGroup Condition="'$(TargetFrameworkIdentifier)' == ''">
    <TargetFrameworkIdentifier Condition="'$(_ShortFrameworkIdentifier)' == 'netstandard'">.NETStandard</TargetFrameworkIdentifier>
    <TargetFrameworkIdentifier Condition="'$(_ShortFrameworkIdentifier)' == 'netcoreapp'">.NETCoreApp</TargetFrameworkIdentifier>
    <TargetFrameworkIdentifier Condition="'$(_ShortFrameworkIdentifier)' == 'net'">.NETFramework</TargetFrameworkIdentifier>
  </PropertyGroup>
```

**Comment by dagood** [comment]
Next step is to get https://github.com/dotnet/core-setup/pull/7324 synced up and through, but with 3.0 and 3.1 work I don't think I'll get to it for at least another week unless it's critical in some way.

**Pull request merged**